src/reflect/scala/reflect/internal/tpe/TypeMaps.scala

```diff
@@ -698,12 +706,21 @@ private[internal] trait TypeMaps {
         tp
     }
 
-    @tailrec private def subst(tp: Type, sym: Symbol, from: List[Symbol], to: List[T]): Type = (
-      if (from.isEmpty) tp
-      // else if (to.isEmpty) error("Unexpected substitution on '%s': from = %s but to == Nil".format(tp, from))
-      else if (matches(from.head, sym)) toType(tp, to.head)
-      else subst(tp, sym, from.tail, to.tail)
-      )
+    private[this] def subst(tp: Type, sym: Symbol): Type = {
+      val found = find(sym)
+      if (found == null) tp else toType(tp, found)
+    }
+
+    protected def find(sym: Symbol): T
+
+    protected def findIndex(sym: Symbol): Int = {
```

Comment:
Why would you use `findIndex` in preference to `findAux`? The call-sites in `find` must index a `List` with the returned index, hence you need two linear scans of the same list, instead of the one in `findAux`.

Reply:
Ah, well, these changes are copied from a previous PR that made a generalisation/abstraction. Under that generalisation, a "substitution map" could be implemented, not as a pair of lists (keys-values), but as a function, or using an array instead of a List (often more compact). In such case, just getting the index, or checking if a symbol is in the "Substitution" domain, would be enough.

But it may not be needed for this PR, though...

Reply:
Ah, makes sense! But here `findIndex` is only called by `find` and used on List. But I'll let you take a look.

Reply:
I have changed this code a bit and dropped the `fromIndex`.